### PR TITLE
For #6211 - Tool Tip for: Privacy and Security Settings page

### PR DESCRIPTION
**Pull request by iorgamgabriel**
This pull request contains the Tool Tip for Privacy and Security Settings page and  Feature Flags for:
 1.Tool Tip for: Privacy and Security Settings page
 2.Home page tips
 3.CFR for Shield toolbar icon

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

**Comment by iorgamgabriel** [comment]
Screenshot_20220204-124505

File: app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt [hunk 1]
```diff
@@ -727,6 +731,8 @@ class UrlInputFragment :
     }
 
     override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String) {
-        if (key == activity?.getString(R.string.pref_key_homescreen_tips)) { updateTipsLabel() }
+        if (key == activity?.getString(R.string.pref_key_homescreen_tips)) {
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt:734]: It would be preferable that the changes not related to this PR, especially indentation or code rearrangements are not included here.

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt [hunk 2]
```diff
@@ -193,11 +201,10 @@ class PrivacySecuritySettingsFragment :
     private fun updateStealthToggleAvailability() {
         val switch =
             preferenceScreen.findPreference(resources.getString(R.string.pref_key_secure)) as? SwitchPreferenceCompat
-        if (preferenceManager.sharedPreferences
-            .getBoolean(
-                    resources.getString(R.string.pref_key_biometric),
-                    false
-                )
+        if (preferenceManager.sharedPreferences.getBoolean(
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt:204]: It would be preferable that the changes not related to this PR, especially indentation or code rearrangements are not included here.

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt [hunk 3]
```diff
@@ -41,6 +43,12 @@ class PrivacySecuritySettingsFragment :
         ) {
             preferenceScreen.removePreference(biometricPreference)
         }
+        if (!Features.SHOULD_SHOW_TOOLTIP_FOR_PRIVACY_SECURITY_SETTINGS_SCREEN ||
+            !requireContext().settings.shouldShowPrivacySecuritySettingsToolTip ||
+            !WhatsNew.shouldHighlightWhatsNew(requireContext())
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt:48]: I think we do not need this check. 
The display conditions described [here](https://github.com/mozilla-mobile/focus-android/issues/6211#issuecomment-1016067914) are covered without it, as the app needs an update to get the new value of the feature flag. 
In turn, if the user does not click the close button, based on UX specs, the CFR should continue to be displayed, even after some time has passed and when the `shouldHighlightWhatsNew` would return false.

File: app/src/main/res/xml/privacy_security_settings.xml [hunk 4]
```diff
@@ -7,6 +7,11 @@
         android:layout="@layout/focus_preference"
         android:summary="@string/preference_privacy_and_security_description"/> -->
 
+    <org.mozilla.focus.settings.privacy.PreferenceToolTipCompose
+        android:key="@string/pref_key_tool_tip"
+        android:layout="@layout/focus_preference_compose_layout"
+        android:selectable="false"/>
+
```
Comment (mcarare) [inline app/src/main/res/xml/privacy_security_settings.xml:14]: I do not see a separator in the Figma file, so we should use ` app:allowDividerBelow="false" ` here

File: buildSrc/src/main/java/Dependencies.kt [hunk 5]
```diff
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 object Versions {
-    const val compose_version = "1.0.4"
+    const val compose_version = "1.0.5"
```
Comment (mcarare) [inline buildSrc/src/main/java/Dependencies.kt:6]: nit: maybe this should be done in a separate commit, if at all in this PR.

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+@Preview
+private fun ToolTipContentPreview() {
+    ToolTipContent {
+    }
+}
+
+/**
+ * Displays a tool tip for Privacy Security Settings screen.
+ *
+ * @param onDismissToolTip Callback when the "x" Button from tool tip is pressed.
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt:47]: nit: I think we can call it the "dismiss button."

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        ) {
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .wrapContentHeight()
+                    .padding(16.dp)
+                    .clip(RoundedCornerShape(10.dp))
+                    .background(
+                        shape = RoundedCornerShape(10.dp),
+                        brush = Brush.verticalGradient(
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt:65]: The Figma designs have a gradient with an angle. This might be possible with a linear gradient with computed offsets or using an XML as `painterResource` for the background.

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+/**
+ * Displays a tool tip for Privacy Security Settings screen.
+ *
+ * @param onDismissToolTip Callback when the "x" Button from tool tip is pressed.
+ */
+@Composable
+fun ToolTipContent(onDismissToolTip: () -> Unit) {
+    FocusTheme {
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt:51]: This should be moved to the topmost composable, in our case the ComposeView. This pattern is followed by the other Compose implementations.

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+                        fontSize = 16.sp,
+                        letterSpacing = 0.5.sp,
+                        color = focusColors.privacySecuritySettingsToolTip,
+                        fontFamily = FontFamily(Font(R.font.metropolis_bold)),
+                        lineHeight = 20.sp
+                    )
+                    Text(
+                        text = stringResource(R.string.tool_tip_message),
+                        fontSize = 14.sp,
+                        letterSpacing = 0.5.sp,
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt:93]: I think the letter spacing is only for the title.
@jeffreygee for confirmation.
Also @jeffreygee , can you confirm the message body text style? It seems it is not the same for all CFR messages. Thank you!

File: app/src/main/java/org/mozilla/focus/utils/CfrUtils.kt [hunk 10]
```diff
@@ -39,41 +39,43 @@ class CfrUtils {
...
-            )
-            if (shieldToolbarIcon != null &&
-                context.settings.shouldShowCfrForShieldToolbarIcon &&
-                isContentSecure
-            ) {
+            if (Features.SHOULD_SHOW_CFR_FOR_SHIELD_TOOLBAR_ICON) {
+                val shieldToolbarIcon = rootView.findViewById<View>(
+                    R.id.mozac_browser_toolbar_tracking_protection_indicator
+                )
+                if (shieldToolbarIcon != null &&
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/utils/CfrUtils.kt:46]: could we move all conditions to a single if expression?

File: app/src/main/java/org/mozilla/focus/utils/Features.kt [hunk 11]
```diff
@@ -9,6 +9,9 @@ package org.mozilla.focus.utils
  */
 object Features {
     const val SEARCH_TERMS_OR_URL: Boolean = true
+    const val SHOULD_SHOW_CFR_FOR_SHIELD_TOOLBAR_ICON: Boolean = false
+    const val SHOULD_SHOW_TOOLTIP_FOR_PRIVACY_SECURITY_SETTINGS_SCREEN: Boolean = false
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/utils/Features.kt:13]: I wonder if we should consider enabling these for debug versions so they can also be tested. @ionutbedregeanu what do you think?

File: app/src/main/java/org/mozilla/focus/settings/privacy/PreferenceToolTipCompose.kt [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+class PreferenceToolTipCompose(context: Context, attrs: AttributeSet?) :
+    Preference(context, attrs) {
+
+    override fun onBindViewHolder(holder: PreferenceViewHolder) {
+        super.onBindViewHolder(holder)
+        val binding = FocusPreferenceComposeLayoutBinding.bind(holder.itemView)
+        binding.toolTipContent.setContent {
+            ToolTipContent {
+                context.settings.shouldShowPrivacySecuritySettingsToolTip = false
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/focus/settings/privacy/PreferenceToolTipCompose.kt:22]: nit: For readability, I think we should keep the lambda in parentheses and maybe also use a named param to avoid any confusion caused by also having a composable here.

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt (hunk 3 above)
Comment (iorgamgabriel) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt]: Done

File: app/src/main/java/org/mozilla/focus/settings/privacy/PreferenceToolTipCompose.kt (hunk 12 above)
Comment (iorgamgabriel) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PreferenceToolTipCompose.kt]: Done

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt (hunk 2 above)
Comment (iorgamgabriel) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsFragment.kt]: CI will not work if I revert the code

File: app/src/main/res/xml/privacy_security_settings.xml (hunk 4 above)
Comment (iorgamgabriel) [reply app/src/main/res/xml/privacy_security_settings.xml:16]: Done

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt (hunk 6 above)
Comment (iorgamgabriel) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt]: Done

File: app/src/main/java/org/mozilla/focus/utils/Features.kt (hunk 11 above)
Comment (ionutbedregeanu) [reply app/src/main/java/org/mozilla/focus/utils/Features.kt:13]: Agree, QA team will thank us

Comment (iorgamgabriel) [reply app/src/main/java/org/mozilla/focus/utils/Features.kt]: Done

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt (hunk 9 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt]: Also, looking at the designs, the gradient has an angle, it's not horizontal or vertical. It might be more difficult obtaining that in Compose (would need a custom element) or you can try having it in an XML and setting it as painter resource.

In either case, @jeffreygee should confirm the result.

**Comment by iorgamgabriel** [comment]
Screenshot_20220207-155213

**Review by mcarare — APPROVED** [review]
Code LGTM. We should wait for the UX feedback on the background gradient.

File: app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt (hunk 9 above)
Comment (jeffreygee) [reply app/src/main/java/org/mozilla/focus/settings/privacy/PrivacySecuritySettingsToolTip.kt]: @mcarare -  since this was a reused component that exists in Fenix right now - is there a way to crosscheck our CFR with theirs?

I was also looking at screenshot - the header looks correct (until we go get rid of Metropolis and completely to native fonts), but the message body should be the the Android native font (Roboto)
